src/setup.js

```diff
@@ -27,9 +50,62 @@ function setupMeasure(dataLayer) {
    * The DataLayerHelper to use with this application.
    * @private @const {!DataLayerHelper}
    */
-  const helper = new DataLayerHelper(dataLayer);
+  const helper = new DataLayerHelper(dataLayer, {'processNow': false});
+
+  helper.registerProcessor('config', configProcessors);
+  helper.process();
+
+  /**
+   * When called with an implementation of the eventProcessor and
+   * storageInterface API, this function registers processors to react to the
+   * 'set' and 'event' commands.
+   *
+   * @param {!ProcessorConstructor|string} eventProcessor The event
+   *     processor to use when responding to event commands
+   *     and deciding how keys should be stored, or a string
+   *     representing an object of this type.
+   * @param {!Object<string, *>} eventOptions Options objects to pass to the
+   *    event processor.
+   * @param {!StorageConstructor|string} storageInterface An
+   *     interface to a class that will handle setting
+   *     and loading long term data, or a string representing
+   *     an object of this type.
+   * @param {!Object<string, *>} storageOptions Options objects to pass to the
+   *    storage interface.
+   */
+  function configProcessors(eventProcessor, eventOptions,
+                            storageInterface, storageOptions) {
+    // TODO: Read in data from the model as extra options
+
+    let processor;
+    let storage;
+    if (typeof eventProcessor === 'string') {
+      if (!(eventProcessor in DEFAULT_EVENT_PROCESSORS)) {
+        log(`No event processor with name ${eventProcessor} found.`,
+            LogLevel.ERROR);
+        return;
+      }
+      processor =
+          new DEFAULT_EVENT_PROCESSORS[eventProcessor](eventOptions);
+    } else {
+      processor = new eventProcessor(eventOptions);
+    }
+    if (typeof storageInterface === 'string') {
+      if (!(storageInterface in DEFAULT_STORAGE_INTERFACES)) {
+        log(`No storage interface with name ${storageInterface} found.`,
+            LogLevel.ERROR);
+        return;
+      }
+      storage =
+          new DEFAULT_STORAGE_INTERFACES[storageInterface](storageOptions);
+    } else {
+      storage = new storageInterface(storageOptions);
+    }
 
-   // TODO: Use the variables declared above
+    /* TODO add event/set processing
```

Comment:
I'm not sure about adding all of this code that essentially no-ops. I think this is a good start to this PR and I'm glad we're starting this review, but I think we might need to make sure the additional functionality is in place before merging this.

Reply:
Well the additional functionality should be coming in #54 and #55!

Reply:
I'd prefer not to wait on those two commits before merging, both of them have some template code to mimic what this commit is because they depend on the variables made in this commit. I could have put all 3 in the same CL but it would have gotten quite large